### PR TITLE
Added docs for background jobs

### PR DESCRIPTION
**Pull request by jacobperia**
## 📜 Description
Sidekiq queues, web interface and priorities.

### CODE REVIEW

File: docs/background_jobs.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+:queues:
+  - [critical, 100]
+  - [default, 50]
+  - [mailers, 25]
+  - [low_priority, 1]
+```
+
+## View queues
+
+Sidekiq also provides a web interface for monitoring job queues. If the necessary environment variables `SIDEKIQ_ADMIN_USERNAME` and `SIDEKIQ_ADMIN_PASSWORD` are set, it can be accessed at `localhost:3000/sidekiq`.
```
Comment (danielpaul) [inline docs/background_jobs.md:21]: give instructions on how they can remove this if they don't want the web dashboard for this. 

say how they can customise the username and password for the login.

File: docs/background_jobs.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Background jobs
+
+Sidekiq is the background job processor we use that handles long-running tasks, such as sending emails or processing images, asynchronously.
```
Comment (danielpaul) [inline docs/background_jobs.md:3]: link to sidekiq gem

File: docs/background_jobs.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+`AttachProfilePictureWorker` is a Sidekiq worker that downloads an image from a given URL and attaches it to the corresponding user record.
```
Comment (danielpaul) [inline docs/background_jobs.md:5]: have a heading for jobs that we currently have in the platform already and this one being one  and what it does

Comment (danielpaul) [reply docs/background_jobs.md:5]: also mention about how the emails by default are sent in the background if deliver_later is used and how sidekiq handles that

File: docs/background_jobs.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
```
Comment (danielpaul) [inline docs/background_jobs.md:6]: mention about how on deployment they need to have another worker for the sidekiq for that process to run and sidekiq to process the queue. also need to mention how redis is required.

**Review by danielpaul — CHANGES_REQUESTED** [review]

File: docs/background_jobs.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
+# Background jobs
+
+[Sidekiq](https://github.com/sidekiq/sidekiq) is the background job processor we use that handles long-running tasks, such as sending emails or processing images, asynchronously.
+
+## Configuration
+
+For sidekiq to work and process jobs, you need to setup two things:
+
+1. A worker `bundle exec sidekiq` needs to be added on your cloud platform and running. This command should be run on your local as well.
+2. Redis is also required. Set it up on your cloud platform as an add-on and set it up on your local with [this tutorial](https://redis.io/docs/getting-started/installation/install-redis-on-mac-os/).
```
Comment (danielpaul) [inline docs/background_jobs.md:10]:
```suggestion
2. Redis for keeping track of the tasks queue. Set it up as a service on your servers and set it up on your local with [this tutorial](https://redis.io/docs/getting-started/installation/install-redis-on-mac-os/).
```

File: docs/background_jobs.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+
+## List of default jobs
```
Comment (danielpaul) [inline docs/background_jobs.md:12]:
```suggestion
## RaidRails Built-in Jobs
```

File: docs/background_jobs.md [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  - [critical, 100]
+  - [default, 50]
+  - [mailers, 25]
+  - [low_priority, 1]
+```
+
+## View queues
+
+Sidekiq also provides a web interface for monitoring job queues which can be accessed at `localhost:3000/sidekiq`
+To enable the web interface, set the username and password in the environment variables `SIDEKIQ_ADMIN_USERNAME` and `SIDEKIQ_ADMIN_PASSWORD` respectively.
```
Comment (danielpaul) [inline docs/background_jobs.md:34]:
```suggestion
To enable the web interface, set the username and password in the environment variables `SIDEKIQ_ADMIN_USERNAME` and `SIDEKIQ_ADMIN_PASSWORD` in the Rails credentials file.
```

File: docs/background_jobs.md [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+:queues:
+  - [critical, 100]
+  - [default, 50]
+  - [mailers, 25]
+  - [low_priority, 1]
+```
+
+## View queues
+
+Sidekiq also provides a web interface for monitoring job queues which can be accessed at `localhost:3000/sidekiq`
```
Comment (danielpaul) [inline docs/background_jobs.md:33]:
```suggestion
Sidekiq also provides a web interface for monitoring job queues which can be accessed at `/sidekiq` on your local and your production server.
```

File: docs/background_jobs.md [hunk 9]
```diff
@@ -0,0 +1,35 @@
+# Background jobs
+
+[Sidekiq](https://github.com/sidekiq/sidekiq) is the background job processor we use that handles long-running tasks, such as sending emails or processing images, asynchronously.
+
+## Configuration
+
+For sidekiq to work and process jobs, you need to setup two things:
+
+1. A worker `bundle exec sidekiq` needs to be added on your cloud platform and running. This command should be run on your local as well.
```
Comment (danielpaul) [inline docs/background_jobs.md:9]:
```suggestion
1. A worker `bundle exec sidekiq` needs to run for the jobs to be processed. 
```

Comment (danielpaul) [reply docs/background_jobs.md:9]: mention how if they are running the server with `bin/dev` then the process run automatically for local development.

**Review by danielpaul — CHANGES_REQUESTED** [review]

File: docs/background_jobs.md (hunk 7 above)
Comment (jacobperia) [reply docs/background_jobs.md:34]: > in the Rails credentials file.

these have been setup as environment variables, not credentials.

File: docs/background_jobs.md [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+  - [critical, 100]
+  - [default, 50]
+  - [mailers, 25]
+  - [low_priority, 1]
+```
+
+## View queues
+
+Sidekiq also provides a web interface for monitoring job queues which can be accessed at `/sidekiq` on your local and your production server.
+To enable the web interface, set the username and password `SIDEKIQ_ADMIN_USERNAME` and `SIDEKIQ_ADMIN_PASSWORD` in the environment variables config.
```
Comment (danielpaul) [inline docs/background_jobs.md:34]:
```suggestion
To enable the web interface, set the username and password `SIDEKIQ_ADMIN_USERNAME` and `SIDEKIQ_ADMIN_PASSWORD` in the environment variables.
```

**Review by danielpaul — APPROVED** [review]

**Pull request merged**